### PR TITLE
Add validation for service entry port.targetPort

### PR DESCRIPTION
**Pull request by ChenglongGao327**
1）if targetport is set, add validation for targetPort
2）move ports parameter validation to the same for loop

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

[ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pkg/config/validation/validation_test.go [hunk 1]
```diff
@@ -4317,6 +4317,37 @@ func TestValidateServiceEntries(t *testing.T) {
 			},
 			valid: false,
 		},
+		{
+			name: "repeat target port", in: networking.ServiceEntry{
+				Hosts:            []string{"google.com"},
+				WorkloadSelector: &networking.WorkloadSelector{Labels: map[string]string{"key": "bar"}},
+				Ports: []*networking.Port{
+					{Number: 80, Protocol: "http", Name: "http-valid1", TargetPort: 80},
+					{Number: 81, Protocol: "http", Name: "http-valid2", TargetPort: 80},
```
Comment (howardjohn) [inline pkg/config/validation/validation_test.go:4326]: This should not be invalid. Its legal in Kubernetes, for example.

**Review by howardjohn — CHANGES_REQUESTED** [review]
I am not sure this is right. Also, we should be careful about adding validation. Any configuration that was previously working on older versions must continue to work or upgrades will be blocked

File: pkg/config/validation/validation_test.go (hunk 1 above)
Comment (ChenglongGao327) [reply pkg/config/validation/validation_test.go:4326]: > This should not be invalid. Its legal in Kubernetes, for example.

I think too much, it should be consistent with k8s

**Review by howardjohn — APPROVED** [review]

**Pull request merged**